### ISSUE TITLE
`rake new_cop` doesn't supports multiple departments

### ISSUE THREAD

**Issue by sue445** [rubocop/rubocop#10117]
I'm writing a rubocop plugin, but passing multiple departments to `new_cop` creates a invalid file

--------

## Expected behavior
When I run`rake new_cop[TopDepartment/SecondDepartment/MyCop]`, the following files will be created

### `lib/rubocop/cop/top_department/second_department/my_cop.rb`
```ruby
# frozen_string_literal: true

module RuboCop
  module Cop
    module TopDepartment
      module SecondDepartment

(omission)
```

### `spec/rubocop/cop/top_department/second_department/my_cop_spec.rb`
```ruby
# frozen_string_literal: true

RSpec.describe RuboCop::Cop::TopDepartment::SecondDepartment::MyCop, :config do
  let(:config) { RuboCop::Config.new }

(omission)
```

## Actual behavior
```bash
$ bundle exec rake new_cop[TopDepartment/SecondDepartment/MyCop]
[create] lib/rubocop/cop/top_department/_second_department/my_cop.rb
[create] spec/rubocop/cop/top_department/_second_department/my_cop_spec.rb
[modify] lib/rubocop/cop/top_department_cops.rb - `require_relative 'top_department/_second_department/my_cop'` was injected.
[modify] A configuration for the cop is added into config/default.yml.
```

### `lib/rubocop/cop/top_department/_second_department/my_cop.rb`
```ruby
# frozen_string_literal: true

module RuboCop
  module Cop
    module TopDepartment/SecondDepartment

(omission)
```

`module TopDepartment/SecondDepartment` is invalid ruby code and `_second_department` (directory name) is also a bit strange.

### `spec/rubocop/cop/top_department/_second_department/my_cop_spec.rb`
```ruby
# frozen_string_literal: true

RSpec.describe RuboCop::Cop::TopDepartment/SecondDepartment::MyCop, :config do
  let(:config) { RuboCop::Config.new }

(omission)
```

`RSpec.describe RuboCop::Cop::TopDepartment/SecondDepartment::MyCop` is invalid ruby code and `_second_department` (directory name) is also a bit strange.

## Steps to reproduce the problem
run `rake new_cop[TopDepartment/SecondDepartment/MyCop]`

## RuboCop version
```bash
$ bundle exec rubocop -V
1.21.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 3.0.2 x86_64-darwin20)
```
